lib/house.rb

```diff
@@ -0,0 +1,50 @@
+class House
+  attr_reader :address, :rooms
+  def initialize(price, address)
+    @price = price
+    @address = address
+    @rooms = []
+    @market_average = 500000
+  end
+
+  def price
+    (@price.delete("$")).to_i
+  end
+
+  def add_room(room)
+    @rooms << room
+  end
+
+  def above_market_average?
+    return true if price > @market_average
+      false
+  end
+
+  def rooms_from_category(category)
+    rooms.find_all do |room|
+      category == room.category
+    end
+  end
+
+  def area
+    room_areas = rooms.map { |room| room.area }
+    room_areas.sum
```

Comment:
this works fine, but `sum` is actually an enumerable too.  So the best solution here would be:

```
def area 
   rooms.sum { |room| room.area }
end
```

Reply:
Good to know!